test/jdk/tools/jpackage/share/InstallDirTest.java

```diff
@@ -113,115 +113,56 @@ public static void testLinuxInvalid(String installDir) {
-
-        DmgTestSpec {
-            Objects.requireNonNull(installDir);
-        }
-
-        static Builder build() {
-            return new Builder();
-        }
-
-        static final class Builder {
-
-            Builder ignoredInstallDir(String v) {
-                installDir = Path.of(v);
-                installDirIgnored = true;
-                return this;
-            }
-
-            Builder acceptedInstallDir(String v) {
-                installDir = Path.of(v);
-                installDirIgnored = false;
-                return this;
-            }
-
-            Builder runtimeInstaller() {
-                runtimeInstaller = true;
-                return this;
-            }
-
-            DmgTestSpec create() {
-                return new DmgTestSpec(installDir, installDirIgnored, runtimeInstaller);
-            }
-
-            private Path installDir;
-            private boolean installDirIgnored;
-            private boolean runtimeInstaller;
-        }
-
-        @Override
-        public String toString() {
-            final var sb = new StringBuilder();
-            sb.append(installDir);
-            if (installDirIgnored) {
-                sb.append(", ignore");
-            }
-            if (runtimeInstaller) {
-                sb.append(", runtime");
-            }
-            return sb.toString();
-        }
-
-        void run() {
-            final var test = new PackageTest().forTypes(PackageType.MAC_DMG).ignoreBundleOutputDir();
-            if (runtimeInstaller) {
-                test.addInitializer(cmd -> {
-                    cmd.removeArgumentWithValue("--input");
-                });
-            } else {
-                test.configureHelloApp();
-            }
-
-            test.addInitializer(JPackageCommand::setFakeRuntime).addInitializer(cmd -> {
-                cmd.addArguments("--install-dir", installDir);
-                cmd.validateOutput(createInstallDirWarningVerifier());
-            }).run(Action.CREATE_AND_UNPACK);
-        }
-
-        private TextStreamVerifier createInstallDirWarningVerifier() {
-            final var verifier = TKit.assertTextStream(
-                    JPackageStringBundle.MAIN.cannedFormattedString("message.install-dir-ignored", defaultDmgInstallDir()).getValue());
-            if (installDirIgnored) {
-                return verifier;
-            } else {
-                return verifier.negate();
-            }
-        }
-
-        private String defaultDmgInstallDir() {
-            if (runtimeInstaller) {
-                return "/Library/Java/JavaVirtualMachines";
-            } else {
-                return "/Applications";
-            }
-        }
+    @Test(ifOS = OperatingSystem.MACOS)
+    @Parameter("/System/jpackage")
+    public static void testMacCreate(String installDir) {
+        new PackageTest()
+        .setExpectedExitCode(1)
+        .excludeTypes(PackageType.MAC_PKG)
```

Comment:
`.forTypes(PackageType.MAC_DMG)` would be more appropriate as the test applies to DMG and not to "all packages except of PKG". If we add another bundle type, `.excludedTypes(Package Type.MAC_PKG)` will become wrong.